### PR TITLE
[Merged by Bors] - doc(library/init/*): document some definitions and theorems

### PR DESCRIPTION
**Pull request by kbuzzard**
A docstring PR adding some docstrings to various mathematically important ideas.

### CODE REVIEW

File: library/init/algebra/classes.lean [hunk 1]
```diff
@@ -71,34 +71,45 @@ class is_idempotent (α : Type u) (f : α → α) : Prop :=
...
 
+/-- `is_antisymm X r` means the binary relation `r` on `X` is antisymmetric. -/
 @[algebra] class is_antisymm (α : Type u) (r : α → α → Prop) : Prop :=
 (antisymm : ∀ a b, r a b → r b a → a = b)
 
+/-- `is_trans X r` means the binary relation `r` on `X` is transitive. -/
 @[algebra] class is_trans (α : Type u) (r : α → α → Prop) : Prop :=
 (trans  : ∀ a b c, r a b → r b c → r a c)
 
+/-- `is_total X r` means that the binary relation `r` on `X` is total, that is, that for any `x y : X` we have `r x y` or `r y x`.-/
```
Comment (bryangingechen) [inline library/init/algebra/classes.lean:102]:
```suggestion
/-- `is_total X r` means that the binary relation `r` on `X` is total, that is, that for any
`x y : X` we have `r x y` or `r y x`.-/
```

File: library/init/algebra/classes.lean [hunk 2]
```diff
@@ -71,34 +71,45 @@ class is_idempotent (α : Type u) (f : α → α) : Prop :=
...
 @[algebra] class is_total (α : Type u) (r : α → α → Prop) : Prop :=
 (total : ∀ a b, r a b ∨ r b a)
 
+/-- `is_preorder X r` means that the binary relation `r` on `X` is a pre-order, that is, reflexive and transitive. -/
```
Comment (bryangingechen) [inline library/init/algebra/classes.lean:106]:
```suggestion
/-- `is_preorder X r` means that the binary relation `r` on `X` is a pre-order, that is, reflexive
and transitive. -/
```

File: library/init/algebra/classes.lean [hunk 3]
```diff
@@ -71,34 +71,45 @@ class is_idempotent (α : Type u) (f : α → α) : Prop :=
...
 
+/-- `is_symm X r` means the binary relation `r` on `X` is symmetric. -/
 @[algebra] class is_symm (α : Type u) (r : α → α → Prop) : Prop :=
 (symm : ∀ a b, r a b → r b a)
 
+/-- The opposite of a symmetric relation is symmetric. -/
 instance is_symm_op_of_is_symm (α : Type u) (r : α → α → Prop) [is_symm α r] : is_symm_op α Prop r :=
 {symm_op := λ a b, propext $ iff.intro (is_symm.symm a b) (is_symm.symm b a)}
 
+/-- `is_asymm X r` means that the binary relation `r` on `X` is asymmetric, that is, `r a b → ¬ r b a`. -/
```
Comment (bryangingechen) [inline library/init/algebra/classes.lean:90]:
```suggestion
/-- `is_asymm X r` means that the binary relation `r` on `X` is asymmetric, that is,
`r a b → ¬ r b a`. -/
```

File: library/init/algebra/classes.lean [hunk 4]
```diff
@@ -71,34 +71,45 @@ class is_idempotent (α : Type u) (f : α → α) : Prop :=
 (idempotent : ∀ a, f (f a) = f a)
 -/
 
+/-- `is_irrefl X r` means the binary relation `r` on `X` is irreflexive (that is, `r x x` never holds). -/
```
Comment (bryangingechen) [inline library/init/algebra/classes.lean:74]:
```suggestion
/-- `is_irrefl X r` means the binary relation `r` on `X` is irreflexive (that is, `r x x` never
holds). -/
```

File: library/init/core.lean [hunk 5]
```diff
@@ -142,6 +142,7 @@ inductive false : Prop
 
 inductive empty : Type
 
+/-- `not P`, with notation `¬ P`, is the Proposition which is true if and only if `P` is false. It is internally represented as `P → false`. -/
```
Comment (robertylewis) [inline library/init/core.lean:145]:
```suggestion
/-- `not P`, with notation `¬ P`, is the `Prop` which is true if and only if `P` is false. It is internally represented as `P → false`. -/
```
to be consistent with below (or change that one)

File: library/init/core.lean [hunk 6]
```diff
@@ -183,6 +184,7 @@ structure prod (α : Type u) (β : Type v) :=
 structure pprod (α : Sort u) (β : Sort v) :=
 (fst : α) (snd : β)
 
+/-- `and P Q`, with notation `P ∧ Q`, is the `Prop` which is true precisely when `P` and `Q` are both true.-/
```
Comment (bryangingechen) [inline library/init/core.lean:187]:
```suggestion
/-- `and P Q`, with notation `P ∧ Q`, is the `Prop` which is true precisely when `P` and `Q` are
both true. -/
```

**Review by bryangingechen — COMMENTED** [review]
Thanks! Just some suggestions breaking up some long lines (though I'm not sure how much we care about applying mathlib style to core Lean).

One thing that would help readability in the generated docs (possibly for a future PR) would be to convert block comments to module comments when appropriate, e.g. all the comments like `/- not -/` and `/- iff -/` in `library/init/logic.lean` could be changed to `/-! ### not -/` and then they would show up as headings.

File: library/init/core.lean (hunk 5 above)
Comment (bryangingechen) [reply library/init/core.lean:145]: I've incorporated this into my suggestion below.

File: library/init/core.lean [hunk 7]
```diff
@@ -142,6 +142,7 @@ inductive false : Prop
 
 inductive empty : Type
 
+/-- `not P`, with notation `¬ P`, is the `Prop` which is true if and only if `P` is false. It is internally represented as `P → false`. -/
```
Comment (bryangingechen) [inline library/init/core.lean:145]:
```suggestion
/-- `not P`, with notation `¬ P`, is the `Prop` which is true if and only if `P` is false. It is
internally represented as `P → false`. -/
```

**Comment by bryangingechen** [comment]
Thanks again!
bors r+

**Pull request closed without merging**